### ISSUE TITLE
[CI] Remove debug build from integration_tests

### ISSUE THREAD

**Issue by andreeaflorescu** [firecracker-microvm/firecracker#1130]
The build takes several minutes and there is no need to check both the release and the debug builds. The debug build can be removed for all platforms.

- update tests/host_tools/cargo_build to run cargo build with --release by default.
- remove `BUILD_TYPES` from tests/integration_tests/build/
- remove `build_type` from `test_build` parameters

There might be other updates needed as well. This needs to be better scoped.

**Comment by McRaeAlex** [comment]
Hi, Apologizes for the mistake of a pull request I am trying to learn how open source contributions work! 

I have completed the required work for this issue over on my fork however do not know how to run the integration tests to ensure that it works as expected. When I made the pull request I know it ran some tests. However I am wondering how I can do it without making the pull request!

Thank you!

**Comment by andreeaflorescu** [comment]
@McRaeAlex you should be able to run the build tests locally on any Linux machine using the container. For functional tests you will need a machine with `/dev/kvm`. To run the build test you need a machine with Docker and then you can run the following command:
```bash
cd tests
../tools/devtool test -- integration_tests/build/test_build.py 
```

**Comment by acatangiu** [comment]
Fixed by #1309
